### PR TITLE
add custom EntityProvider to create catalog backstage entities from k…

### PR DESCRIPTION
**Pull request by ca7alindev**
…ubernetes objects

### CODE REVIEW

File: packages/backstage/backend/src/plugins/EntityProvider.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  private readonly env: string;
+  private readonly reader: UrlReader;
+  private connection?: EntityProviderConnection;
+
+  private readonly apiEndpoints: string[]; // Define API endpoints here
+
+  constructor(env: string, reader: UrlReader) {
+    this.env = env;
+    this.reader = reader;
+    const defaultEndpoints = [ 'https://kubernetes.default.svc/apis/kndp.io/v1alpha1/releases' ];
```
Comment (web7evgheni) [inline packages/backstage/backend/src/plugins/EntityProvider.ts:16]: @catalindev0ps No default endpoints, everything shall come from environment!!!

File: packages/backstage/backend/src/plugins/EntityProvider.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    const entities: Entity[] = [];
+    const parsedData = JSON.parse(data);
+    if (parsedData) {
+      const entity = {
+        kind: 'Component',
+        apiVersion: 'backstage.io/v1alpha1',
+        metadata: {
+          annotations: {
+            "backstage.io/managed-by-location": "file:../../../..//catalog-info.yaml",
+            "backstage.io/managed-by-origin-location": "file:../../../../catalog-info.yaml"
```
Comment (web7evgheni) [inline packages/backstage/backend/src/plugins/EntityProvider.ts:83]: @catalindev0ps Why this is required?

File: packages/backstage/charts/backstage-backend/templates/deployment.yaml [hunk 3]
```diff
@@ -28,7 +28,13 @@ spec:
...
+          env:
+          - name: API_ENDPOINTS
+            value: {{- .Values.apiEndpoints | join(";") -}}
           ports:
             - name: http
               containerPort: {{ .Values.service.port }}
               protocol: TCP
+      envFrom:
+      - secretRef:
+          name: backstage-token
```
Comment (web7evgheni) [inline packages/backstage/charts/backstage-backend/templates/deployment.yaml:40]: @catalindev0ps Use service account token. And if required add RBAC configuration for this service account for read all current API groups we have in KNDP

**Review by web7evgheni — CHANGES_REQUESTED** [review]

File: packages/backstage/backend/src/plugins/EntityProvider.ts (hunk 2 above)
Comment (catalindev0ps) [reply packages/backstage/backend/src/plugins/EntityProvider.ts:83]: got "catalog warn Entity 'component:default/release' does not have the annotation backstage.io/managed-by-location type=plugin entity=component:default/crossplane location=undefined" and without this annotations entity is not displayed in frontend

File: packages/backstage/charts/backstage-backend/templates/deployment.yaml (hunk 3 above)
Comment (web7evgheni) [reply packages/backstage/charts/backstage-backend/templates/deployment.yaml:40]: @catalindev0ps This backstage-token still required?

File: packages/backstage/backend/src/plugins/EntityProvider.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        spec: {
+          type: parsedData.kind,
+          lifecycle: 'experimental',
+          owner: 'guests'
+        },
+      };
+
+      entities.push(entity);
+      console.log(entity);
```
Comment (web7evgheni) [inline packages/backstage/backend/src/plugins/EntityProvider.ts:91]: @catalindev0ps Remove console.log

File: packages/backstage/charts/backstage-backend/templates/rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  name: {{ .Chart.Name }}-sa
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Chart.Name }}-cr
+rules:
+  - apiGroups: ['kndp.io']
+    resources: ['releases']
```
Comment (web7evgheni) [inline packages/backstage/charts/backstage-backend/templates/rbac.yaml:13]: @catalindev0ps  please use resources: ["*"] to be possible to access next resources types.

File: packages/backstage/charts/backstage-backend/templates/rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    verbs: ['get', 'list', 'watch', 'create', 'edit']
```
Comment (web7evgheni) [inline packages/backstage/charts/backstage-backend/templates/rbac.yaml:14]: @catalindev0ps Please add delete also

**Review by web7evgheni — CHANGES_REQUESTED** [review]

File: packages/backstage/backend/src/plugins/EntityProvider.ts (hunk 4 above)
Comment (ca7alindev) [reply packages/backstage/backend/src/plugins/EntityProvider.ts:91]: @web7evgheni yes, i think this "file:../../../..//catalog-info.yaml" should be replaced with the location from where EntityProvider fetch data, like "https://kubernetes.default.svc"

Comment (ca7alindev) [reply packages/backstage/backend/src/plugins/EntityProvider.ts:91]: As is explain in docs "The value of this annotation is a so called location reference string, that points to the source from which the entity was originally fetched."

Comment (web7evgheni) [reply packages/backstage/backend/src/plugins/EntityProvider.ts:91]: @ca7alindev Proof link please.

Comment (ca7alindev) [reply packages/backstage/backend/src/plugins/EntityProvider.ts:91]: https://backstage.io/docs/features/software-catalog/well-known-annotations/

Comment (web7evgheni) [reply packages/backstage/backend/src/plugins/EntityProvider.ts:91]: @ca7alindev Make it link to fetched URL, means k8s api.

**Review by web7evgheni — CHANGES_REQUESTED** [review]

**Review by web7evgheni — APPROVED** [review]

**Pull request merged**